sql: SHOW CREATE ALL SCHEMAS doesn't work with comment on

hanpen24 opened the pull request:
There was a bug where comments did not show up in
  the output of the show create all schemas command.
  The issue was that we did not have any code to get
   the comment and add it to the create schema
   descriptor.

  Fixes: #127979

  Release note (bug fix): SHOW CREATE ALL SCHEMAS
  now shows corresponding schema comments in it's output.

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=130164) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=130164) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=130164) it.</sub>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/130164)
<!-- Reviewable:end -->

Dedej-Bergin reviewed (commented):
Great work! Just one minor comment

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130164#-:-O62H2tx2rIEWWS1KzJk:bti9kj2)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @hanpen24)

___
*[`pkg/sql/crdb_internal.go` line 3625 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130164#-O62GdQf3Rloa3pwk3an:-O62GdQg-Ef3LOEWi1bp:b-3d6vhk) ([raw file](https://github.com/cockroachdb/cockroach/blob/868533cef781ba4e27a18ee7a15a078e6eec58ce/pkg/sql/crdb_internal.go#L3625)):*
> ```Go
> 					comment, ok := p.Descriptors().GetSchemaComment(schemaDesc.GetID())
> 					if ok {
> 						commentOnType := tree.CommentOnSchema{Comment: &comment, Name: tree.ObjectNamePrefix{SchemaName: tree.Name(schemaDesc.GetName()), ExplicitSchema: true}}
> ```

small typo, this should be called commentOnSchema


<!-- Sent from Reviewable.io -->

Dedej-Bergin commented:
@hanpen24 I see the licensed/cla is not signed yet, let me know if you need any help with it.

Dedej-Bergin commented:
@hanpen24 for the failed unit test it may just need a rebase with master:

Run these git commands to rebase:

git checkout master
git pull
git checkout my-feature
// do not do git merge
git rebase master --interactive
// interactive can help squash/edit multiple commits
// solve merge conflicts if any
// git add <modified file>
// git rebase –continue
git push --force hanpen24

Dedej-Bergin commented:
@hanpen24 for the failed unit test add nosort to line 47 like this:

```
 47 query T colnames,nosort
 48 SHOW CREATE ALL SCHEMAS
 49 ----
 50 create_statement
 51 CREATE SCHEMA public;
 52 COMMENT ON SCHEMA public IS 'test-comment';
 53 CREATE SCHEMA test2;
 ```

hanpen24 commented:
@Dedej-Bergin 
Thank you for the review! 
I have fixed the typos and updated the unit tests.
Also, I have signed the licensed/CLA.
I am looking forward to your review.

hanpen24 commented:
I will check the failed tests

hanpen24 commented:
The error was the same as the one in the following PR: https://github.com/cockroachdb/cockroach/pull/129983, so I merged master.

hanpen24 committed:
sql: SHOW CREATE ALL SCHEMAS doesn't work with comment on
There was a bug where comments did not show up in
the output of the show create all schemas command.
The issue was that we did not have any code to get
the comment and add it to the create schema descriptor.

Fixes: #127979

Release note (bug fix): SHOW CREATE ALL SCHEMAS
now shows corresponding schema comments in it's output.

hanpen24 commented:
The error was the same as the one in the following PR: https://github.com/cockroachdb/cockroach/pull/129983, so I merged master.

Dedej-Bergin reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

Great work! Thanks for your contribution @hanpen24 !!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130164#-:-O7ZC-NvEoruYsyud074:b17ob93)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @hanpen24)



<!-- Sent from Reviewable.io -->

Dedej-Bergin commented:
bors r+
